### ISSUE TITLE
 Could not load assembly NuGet.Core

### ISSUE THREAD

**Issue by Silvenga** [cake-build/cake#623]
This looks similar to #549. 

Under Mono, CakeEnvironment appears to be throwing a missing method exception on the getter `TargetFrameworkName` (line 133) - under version 3.2.8 (Debian 3.2.8+dfsg-4ubuntu1.1) from the official repositories for Ubuntu 14.04.

```
Missing method System.AppDomainSetup::get_TargetFrameworkName() in assembly /usr/lib/mono/4.5/mscorlib.dll, referenced in assembly /tmp/btsync-deb/tools/Cake/Cake.Core.dll
Error: Method not found: 'System.AppDomainSetup.get_TargetFrameworkName'.
```

**Comment by devlead** [comment]
Which version of Cake and Mono are you running?

Fix for #549 came with 0.6.2 but if not already running it I recommend try latest as allot been added and bugs fixed.

Currently we've only tested on Mono 4, If you're not running 4, I recommend that you fetch 4.2.1 from Xamarin’s servers. It has allot of bug fixes.
You can find instructions [here](http://www.mono-project.com/docs/getting-started/install/linux/#debian-ubuntu-and-derivatives)

**Comment by Silvenga** [comment]
Upgrading Mono did fix the missing method exception, however now I get the following under Cake 0.7.0. 

```
Error: Could not load file or assembly 'NuGet.Core, Version=2.8.60717.93, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.
```

I originally upgraded from Cake 0.5.5 - so I'm thinking I will revert and try again sometime in the future.

**Comment by patriksvensson** [comment]
@silvenga Did you remove the Cake folder completely when upgrading? You could see this error if an old NuGet.Core exists in the directory.

**Comment by Silvenga** [comment]
My build server should clean the repository before building. To make sure of this, I tried again with a manual fresh clone - same issue. 

Backtracking through each version I saw this issue first appeared in v0.6.2

0.6.0 - no issues
0.6.1 - `Error: Value cannot be null. Parameter name: frameworkName`
0.6.2 - `Error: Could not load file or assembly 'NuGet.Core, Version=2.8.60717.93, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.`
0.6.3 - `Error: ApplicationName='"/usr/bin/nuget"', CommandLine='install "Cake.FileHelpers" -OutputDirectory "/tmp/deb/tools/Addins" -ExcludeVersion -NonInteractive -NoCache', CurrentDirectory='/tmp/deb', Native error= Cannot find the specified file`
0.6.4 - `Error: Could not load file or assembly 'NuGet.Core, Version=2.8.60717.93, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.`

I modified the `packages.conf` setting the cake version. I then ran the build with `git clean -fXd && bash build.sh`

**Comment by patriksvensson** [comment]
@silvenga That's strange. And v0.7.0?

**Comment by Silvenga** [comment]
> And v0.7.0?

Yep.

**Comment by patriksvensson** [comment]
@Silvenga Does this only happen on your build machine? Can you recreate the behavior locally?

**Comment by Silvenga** [comment]
The same behavior occurs on my home server (the first Linux machine I thought of). I also had to install Nuget into my path (it was complaining about not finding nuget.exe).

**Comment by nathansoz** [comment]
I encountered this issue. On my box, it turned out that tools/Cake/NuGet.Core.dll was actually cased as tools/Cake/Nuget.Core.dll. That capitalized "G" matters on a case sensitive file system :)

**Comment by Silvenga** [comment]
I can confirm that this is resolved in Cake v0.10.1. Thanks @nathansoz!

**Comment by devlead** [comment]
Superb :+1:
